Mathlib/CategoryTheory/Category/Preorder.lean

```diff
@@ -171,28 +168,66 @@ def OrderIso.equivalence (e : X ≃o Y) : X ≌ Y where
 
 end
 
-namespace CategoryTheory
-
 section Preorder
 
 variable {X : Type u} {Y : Type v} [Preorder X] [Preorder Y]
 
-/-- A functor between preorder categories is monotone.
--/
+namespace CategoryTheory.Functor
+
+/-- A functor between preorder categories is monotone. -/
 @[mono]
-theorem Functor.monotone (f : X ⥤ Y) : Monotone f.obj := fun _ _ hxy => (f.map hxy.hom).le
+theorem monotone (f : X ⥤ Y) : Monotone f.obj := fun _ _ hxy => (f.map hxy.hom).le
+
+/-- A functor `X ⥤ Y` between preorder categories as an `OrderHom`. -/
+@[simps!]
+def toOrderHom (F : X ⥤ Y) : (X →o Y) where
+  toFun := F.obj
+  monotone' := F.monotone
+
+end CategoryTheory.Functor
+
+namespace OrderHom
+
+open CategoryTheory
+
+/-- An `OrderHom` as a functor `X ⥤ Y` between preorder categories. -/
+abbrev toFunctor (f : X →o Y) : X ⥤ Y := f.monotone.functor
+
+/-- The equivalence between `X →o Y` and the type of functors `X ⥤ Y` between preorder categories
+`X` and `Y`. -/
+@[simps]
+def equivFunctor : (X →o Y) ≃ (X ⥤ Y) where
+  toFun := toFunctor
+  invFun F := F.toOrderHom
+
+/-- The categorical equivalence beween the category of monotone functions `X →o Y` and the category
+of functors `X ⥤ Y`, where `X` and `Y` are preorder categories. -/
+@[simps! functor_obj_obj inverse_obj unitIso_hom_app unitIso_inv_app counitIso_inv_app_app
+  counitIso_hom_app_app]
+def equivalenceFunctor : (X →o Y) ≌ (X ⥤ Y) where
+  functor :=
+    { obj f := f.toFunctor
+      map f := { app a := f.down.down a |>.hom } }
+  inverse :=
+    { obj F := F.toOrderHom
+      map f := ⟨⟨fun i ↦ f.app i |>.down.down⟩⟩ }
```

Comment:
It is better not to rely on the implementation details of the category structure on a preorder:
```suggestion
  functor :=
    { obj f := f.toFunctor
      map f := { app a := homOfLE (leOfHom f _) } }
  inverse :=
    { obj F := F.toOrderHom
      map f := homOfLE (fun i ↦ leOfHom (f.app i) ) }
```